### DISCUSSION TITLE
Today's Detections broken for me?

### DISCUSSION THREAD

**Discussion by SereneSloth** [Nachtzuster/BirdNET-Pi#169] [Q&A] [answered]
Hey all - I am having the same issue as a similar post on here but never saw a resolution beyond it just started working again.

I've had 2 days in a row where after a certain time my Today's Detections are failing to load correctly. They show the entry, but no spectrograph or no ability to play audio. Looking at the network tab in the browser it is saying it cannot find the file, however the file exists, for sure. See screenshots for a specific example but they all do this same behavior. It seems to resolve after a new day, but at some point, in the day it reverts back to this behavior.

If anyone has any recommendations or needs more info let me know.

This is showing what it looks like on the page:
![msedge_q5LkJBsrRG](https://github.com/user-attachments/assets/ae54c1b0-2cb6-4f28-9a61-619e98ab6857)

This shows the errors in the browser network log:
![msedge_1Z1qKCs75q](https://github.com/user-attachments/assets/1854305a-e7e6-45a0-b09e-1c9e22dab0a1)

This shows the file exists within the file manager in the birdnet webui:
![msedge_78VJxZAKi5](https://github.com/user-attachments/assets/9af6b93c-5d67-49ce-9771-ec35b98f4637)

This shows the file exists from a terminal view:
![WindowsTerminal_d8aYYJi1kX](https://github.com/user-attachments/assets/c36a4f7f-2ce4-4bb2-b621-bc72a5d70fd2)

**Comment by SereneSloth** [comment]
Something interesting I've noticed after posting, the network tab in the browser is showing it's looking for the URL with the following day's date. It is currently 2024-09-11, and it appears to be looking for a file in 2024-08-12. Not sure why that is happening. My system clock shows the correct time, and so does the settings tab within birdnet.

![WindowsTerminal_STvHwr7nvf](https://github.com/user-attachments/assets/8a81a8c5-763f-4433-9556-3223c868ad21)

**Comment by Emmo213** [comment]
Next time this starts happening take a look at the log. You should see a line that says something like "Analyzing BirdSongs/StreamData/2024-08-12-birdnet-12:57:00.wav". Is the date and time correct on that filename, or is the date tomorrow? That's what drives all the other date and timestamps.

**Reply by SereneSloth** [reply]
Will do. I didn't think to check that when it was doing it sadly. It is - as expected - fine as of 12am on Monday and still rocking on as of 2:26pm CDT. Timestamps currently seem to be correct in the logs.

![msedge_JWJQElFlso](https://github.com/user-attachments/assets/3f9e0962-3bc7-45aa-a126-9efd60df3f86)

I'll report back if/when it messes up again and see if anything changes.

**Reply by SereneSloth** [reply]
Sadly, same thing has happened again. The entire page is messed up, in the browser it is looking for the next day's folder while it is still processing today's date.

![msedge_dKQEH1oWPI](https://github.com/user-attachments/assets/eef723ba-979a-4b50-b342-ca04faccb226)
![msedge_mr9kZ883c9](https://github.com/user-attachments/assets/afa25c09-bfab-422b-b6c6-009b6ace2ae3)
![msedge_HWlCL0YFED](https://github.com/user-attachments/assets/0e41782f-11a9-4091-bb45-a7a56d8e66f1)

**Reply by SereneSloth** [reply]
Another interesting thing I found is that whatever the issue is, may be related to the caddy service. Given it is showing a GMT timezone in the response header.

![msedge_imzcmg6Joz](https://github.com/user-attachments/assets/f133d958-65bf-4020-9c79-aa6eb64b88b2)

**Comment by Emmo213** [comment]
I rarely use the toadys detections but mine was doing it too. In todays_detections.php I added set_timezone() under the site name and it works for me now. 

<img width="218" alt="image" src="https://github.com/user-attachments/assets/a09d1508-0411-4796-8c62-59dd4f7e8c20">

Looks like an actual bug. This should probably be moved from discussion to issues.

**Reply by SereneSloth** [reply]
You're the best, adding that line has fixed it for me. I appreciate your time.

Do you need me to move this into the issues section or is that something you can do?

**Reply by Emmo213** [reply]
I added it, and will submit a PR tomorrow for the fix to get pulled into the main branch. 

https://github.com/Nachtzuster/BirdNET-Pi/issues/170

**Reply by SereneSloth** [reply]
Awesome. Thank you so much for your help.

**Comment by Emmo213** [comment, marked as answer]
This was merged into the main branch this morning. If possible please mark this discussion as answered.
